service/matching/task_reader.go

```diff
@@ -51,11 +50,9 @@ const (
 
 type (
 	taskReader struct {
-		status     int32
 		taskBuffer chan *persistencespb.AllocatedTaskInfo // tasks loaded from persistence
 		notifyC    chan struct{}                          // Used as signal to notify pump of new tasks
 		backlogMgr *backlogManagerImpl
-		gorogrp    goro.Group
```

Comment:
❤️

Reply:
I don't _hate_ goro.Group/goro.Handle, but I'm souring on them slightly: they package up a few common patterns, but encapsulate things so that they're not composable with themselves or other startup/shutdown patterns.

Reply:
I haven't used it myself, but using the Go primitives instead where possible seems like a great change.